.github/workflows/ci.yml

```diff
@@ -230,6 +230,57 @@ jobs:
           if-no-files-found: error
           retention-days: 14
 
+  Unit-Test-Suite:
+    needs: [Precheck, Checkout]
+    environment: Testing
+    runs-on: ubuntu-latest
+    if: needs.precheck.outputs.srcfileschanged == 'true'
+    defaults:
+      run:
+        working-directory: src
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Golang with cache
+        uses: magnetikonline/action-golang-cache@v4
+        with:
+          go-version-file: src/go.mod
+
+      - run: mkdir testlog
+
+      # Install gotestfmt
+      - name: Set up gotestfmt
+        run: go install github.com/gotesttools/gotestfmt/v2/cmd/gotestfmt@latest
+
+      # run the tests
+      - name: Unit Tests
```

Comment:
Wondering if we could make `Test-Suite-Trusted` run non unit tests alone so as to not run them twice.